### PR TITLE
Feature/p22 46 list timeslots

### PR DESCRIPTION
**Pull request by rapaicd**

### CODE REVIEW

File: tennis-scheduler-web-app/src/components/Login.jsx [hunk 1]
```diff
@@ -11,14 +11,15 @@ export const Login = () => {
   const loginSubmit = e => {
     e.preventDefault();
     setFormErrors(validation(formValues));
-
+    //there will be login request, and loginMessage will be set if login is failed
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Login.jsx]: Delete this

File: tennis-scheduler-web-app/src/components/Login.jsx [hunk 2]
```diff
@@ -11,14 +11,15 @@ export const Login = () => {
...
     if (Object.keys(formErrors).length === 0) {
       login(formValues).then((response) => {
         localStorage.setItem("token", response.data.accessToken);
         navigate('/profile');
         window.location.reload();
       }).catch(() => setLoginMessage("Wrong credencials, try again!"))
     }
+
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Login.jsx]: No need for whitespace

File: tennis-scheduler-web-app/src/components/Navbar.jsx [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
+  }
+
 
 
 export const NavbarStart = () => {
   const [tennisPlayer, setTennisPlayer] = useState(false);
   const [admin, setAdmin] = useState(false);
   const navigation = useNavigate();
 
+
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Navbar.jsx]: No need for whitespace

File: tennis-scheduler-web-app/src/components/Navbar.jsx [hunk 4]
```diff
@@ -42,6 +54,20 @@ export const NavbarStart = () => {
...
+              Login
+            </Link>
+            <Link className="nav-link" to="/registration">
+              Register
+            </Link>
+            {admin ? <span>
+            <Link className="nav-link" to="/timeslots">
+              Timeslots
+            </Link></span> : ""}
+          </Nav>
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Navbar.jsx]: There is already admin section

File: tennis-scheduler-web-app/src/components/Timeslot.jsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import UpdateTimeslot from "./UpdateTimeslot";
+import { deleteTimeslot } from "../api/TimeslotApi";
+import ListGroup from 'react-bootstrap/ListGroup';
+import Button from 'react-bootstrap/Button';
+import moment from "moment/moment";
+import '../styles/timeslot.css'
+
+
+const Timeslot = ({ timesl, setTimeslots, timeslots}) => {
+  console.log(timesl.id)
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Timeslot.jsx]: Delete this and change name of variable timesl

File: tennis-scheduler-web-app/src/components/Timeslot.jsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  useEffect(() => {
+    getTennisCourts().then((data) => {
+      setTennisCourts(data.data);
+    })
+  }, []);
+  useEffect(() => {
+    getPerson().then((data) => {
+      setPerson(data.data);
+    })
+  }, []);
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/Timeslot.jsx]: What if request fail? Add catch and show user error

File: tennis-scheduler-web-app/src/components/UpdateTimeslot.jsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+            "personId": timeslot.personId,
+            "courtId": timeslot.tennisCourtId
+        };
+
+        updateTimeslot(id, newTimeslot).then(() => {
+            setTimeslotErrors("");
+        })
+            .catch((errorMessage) => {
+                setTimeslotErrors(errorMessage.response.data.message[0].defaultMessage);
+            })
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/components/UpdateTimeslot.jsx]: Fix indentation

File: tennis-scheduler-web-app/src/pages/Timeslots.jsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+  const [timeslots, setTimeslots] = useState([]);
+
+
+  useEffect(() => {
+    getTimeslot().then((data) => {
+      setTimeslots(data.data);
+    })
+  }, []);
+  console.log(timeslots[1])
```
Comment (markoacimovic) [inline tennis-scheduler-web-app/src/pages/Timeslots.jsx]: Delete this

Comment (markoacimovic) [reply tennis-scheduler-web-app/src/pages/Timeslots.jsx]: !

File: tennis-scheduler-web-app/src/components/UpdateTimeslot.jsx (hunk 7 above)
Comment (markoacimovic) [reply tennis-scheduler-web-app/src/components/UpdateTimeslot.jsx]: !

File: tennis-scheduler-web-app/src/components/UpdateTimeslot.jsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+import { getTimeslot } from "../api/TimeslotApi";
+import { updateTimeslot } from '../api/TimeslotApi'
```
Comment (cyberpoetry17) [inline tennis-scheduler-web-app/src/components/UpdateTimeslot.jsx]: Can be single line.

**Review by markoacimovic — APPROVED** [review]

**Pull request merged**